sdk/core/core/test/cmocka/test_span.c

```diff
@@ -250,42 +250,32 @@ static void az_span_find_beginning_success()
 {
   az_span span = AZ_SPAN_FROM_STR("abcdefgabcdefg");
   az_span target = AZ_SPAN_FROM_STR("abc");
-  az_span instance;
 
-  assert_return_code(az_span_find(span, target, &instance), AZ_OK);
-  assert_true(az_span_length(target) == az_span_length(instance));
-  assert_true(az_span_ptr(span) == az_span_ptr(instance));
+  assert_int_equal(az_span_find(span, target), 0);
 }
 
 static void az_span_find_middle_success()
 {
   az_span span = AZ_SPAN_FROM_STR("abcdefgabcdefg");
   az_span target = AZ_SPAN_FROM_STR("gab");
-  az_span instance;
 
-  assert_return_code(az_span_find(span, target, &instance), AZ_OK);
-  assert_true(az_span_length(target) == az_span_length(instance));
-  assert_true((az_span_ptr(span) + 6) == az_span_ptr(instance));
+  assert_int_equal(az_span_find(span, target), 6);
 }
 
 static void az_span_find_end_success()
 {
   az_span span = AZ_SPAN_FROM_STR("abcdefgabcdefgh");
   az_span target = AZ_SPAN_FROM_STR("efgh");
-  az_span instance;
 
-  assert_return_code(az_span_find(span, target, &instance), AZ_OK);
-  assert_true(az_span_length(target) == az_span_length(instance));
-  assert_true((az_span_ptr(span) + 11) == az_span_ptr(instance));
+  assert_int_equal(az_span_find(span, target), 11);
 }
 
 static void az_span_find_not_found_fail()
 {
   az_span span = AZ_SPAN_FROM_STR("abcdefgabcdefg");
   az_span target = AZ_SPAN_FROM_STR("abd");
-  az_span instance;
 
-  assert_true(az_span_find(span, target, &instance) == AZ_ERROR_ITEM_NOT_FOUND);
+  assert_int_equal(az_span_find(span, target), -1);
```

Comment:
Or this is the symbol. I think I'm tempted to just leave it as -1 and not define the enum symbol

Reply:
i think i agree with the -1 as a number. I believe that's common with other languages too

Reply:
We can do that. Let's keep it simple. I'll change it to "-1"